CPP_STYLE_GUIDE.md

```diff
@@ -315,8 +315,10 @@ exports.foo = function(str) {
 
 #### Avoid throwing JavaScript errors in nested C++ methods
 
-When you need to throw a JavaScript exception from C++, try to do it at the top
-level and not inside of nested calls.
+When you need to throw a JavaScript exception from C++ (i.e.
+`isolate()->ThrowException()`) prefer to do it as close to the return to JS as
+possible, and not inside of nested C++ calls. Since this changes the `isolate`'s
+state doing closest to where it is consumed, reduces the chance of side effects.
```

Comment:
@joyeecheung, @addaleax PTAL
I'm trying to grok this, so I wrote it down.

Reply:
s/isolate/Isolate/, and the comma should be removed

I don’t think it matters that it’s an `Isolate` thing, though. That’s implementation details, and you can just say “JS execution state” or something like that.

Reply:
Ack